src/index.ts

```diff
@@ -10,26 +19,75 @@ export async function sync(name: string, from: Record<string, string>, to: Recor
   // fullMetadata may needed to obtain the repository property in manifest
   srcOptions.fullMetadata = true
   const srcPackument = await getPackument(name, srcOptions)
-  const srcVersions = srcPackument ? Object.keys(srcPackument.versions) : []
-  console.debug('Source versions', srcVersions)
+  const srcVersionsRaw = srcPackument ? Object.keys(srcPackument.versions) : []
+
+  console.debug('Source versions', srcVersionsRaw)
+
+  let srcVersions = srcVersionsRaw;
+
+  if (srcPackument && latestOnly) {
+    srcVersions = [srcPackument['dist-tags'].latest]
+    console.debug('Latest version', srcVersions)
+  } else if (srcVersionsRaw.length && latestMajors) {
+    const removePrereleases = srcVersionsRaw.filter(version => {
+      const semverInstance = new semver.SemVer(version)
+      return semverInstance.prerelease.length === 0;
+    });
+
+    srcVersions = Array.from(semver.rsort(removePrereleases).reduce((acc, version) => {
+      const semverInstance = new semver.SemVer(version)
+
+      if (semverInstance.major === 0 && semverInstance.minor === 0) {
+        // every one
+        acc.set(version, version)
+      } else if (semverInstance.major === 0) {
+        const key = `${semverInstance.major}.${semverInstance.minor}`
+        if (!acc.has(key)) acc.set(key, version.toString())
+      } else if (!acc.has(semverInstance.major)) {
+        acc.set(semverInstance.major, version.toString())
+      }
+
+      return acc;
+    }, new Map).values()).reverse() as string[]
+
+    console.debug('Latest major versions', srcVersions)
+  }
 
   // get available target versions
   const dstOptions = scopedOptions(scope, to.registry, to.token)
   const dstPackument = await getPackument(name, dstOptions)
   const dstVersions = dstPackument ? Object.keys(dstPackument.versions) : []
   console.debug('Target versions', dstVersions)
 
-  const missing = srcVersions.filter(x => !dstVersions.includes(x))
+  let missing = srcVersions.filter(x => !dstVersions.includes(x))
```

Comment:
Codacy has a fix for the issue: [Replace `x·=>·!dstVersions.includes(x))` with `(x)·=>·!dstVersions.includes(x));`](https://app.codacy.com/gh/joebowbeer/regsync/pullRequest?prid=9263163)
```suggestion
  let missing = srcVersions.filter((x) => !dstVersions.includes(x));
```